tiny_dnn/core/framework/tensor.h

```diff
@@ -128,16 +128,22 @@ class TensorStorage {
  */
 template<typename U = float_t, size_t kDimensions = 4, bool kConst = false>
```

Comment:
I'm not sure I get you right. 
My idea is, that Tensor can be forced to be constant at compile time, i.e. incapable to modify underlying TensorStorage (surely mutable).